### PR TITLE
[#3861] IBM Watson assistant connector backend

### LINKED ISSUE #3861: IBM Watson Assistant Connector in the backend
-> add the IBM Watson Assistant Connector in the backend

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #3861

### CODE REVIEW

File: backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/IbmWatsonAssistantClientConfig.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                .client(new OkHttpClient())
+                .encoder(new JacksonEncoder())
+                .decoder(new JacksonDecoder())
+                .requestInterceptor(new BasicAuthRequestInterceptor("apikey", apiKey))
+                .logger(new feign.Logger.ErrorLogger())
+                .logLevel(feign.Logger.Level.FULL)
+                .target(IbmWatsonAssistantClient.class,
+                        String.format("%1$s/v2/assistants/%2$s/message?version=2021-06-14", url, assistantId));
+    }
+}
```
Comment (AudreyKj) [inline backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/IbmWatsonAssistantClientConfig.java:28]: - Basic auth is needed to authenticate to the API
- The request URL is comprised of the Ibm Watson Instance URL + assistantId

File: backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    public String getContent(String source, MessageSendResponse response) throws JsonProcessingException {
+
+        try {
+            final JsonNode responseNode = mapper.valueToTree(response);
+            String text = "";
+
+            for (JsonNode nestedNode : responseNode.path("output").path("generic")) {
+                text = nestedNode.get("text").textValue();
+            }
+
```
Comment (AudreyKj) [inline backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java:56]: in the response from the AI assistant, the text is nested inside of the `output.generic` array.

File: backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    }
+                    case "whatsapp": {
+                        node.put("Body", text);
+                        return mapper.writeValueAsString(node);
+                    }
+
+                    default: {
+                        return null;
+                    }
+                }
```
Comment (juan-sebastian) [inline backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java:91]: We should move this to a library, in a other PR so we can use it in all this times of connectors, reducing repetitive code

**Review by juan-sebastian — APPROVED** [review]
Great work 💪

File: backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java (hunk 3 above)
Comment (AudreyKj) [reply backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java:91]: yes good point 👍  I'm happy to take care of this! Created the ticket: #3890

**Pull request merged**